### ISSUE TITLE
issue with parsing float in MS Windows

### ISSUE THREAD

**Issue by sl-solution** [JuliaLang/julia#45612]
The `parse` function performs poorly in MS Windows, and using simple threading escalates the problem significantly. This issue has been opened in `DLMReader.jl` by @giantmoa, however, it is related to `Julia`. Please see the following links for more information:

https://github.com/sl-solution/DLMReader.jl/issues/5

**Comment by inkydragon** [comment]
Related #24015

**Comment by metab0t** [comment]
The implementation of `jl_strtod_c` is in https://github.com/JuliaLang/julia/blob/master/src/support/strtod.c
Locale is not cached on Windows. It might influence the performance.

**Comment by metab0t** [comment]
BTW, if you need performance, Parsers.jl should be the first choice.
`Parsers.parse` is way more faster than `Base.parse`.

```julia
julia> @btime Parsers.parse(Float64, "3.1415926")
  14.615 ns (0 allocations: 0 bytes)
3.1415926

julia> @btime parse(Float64, "3.1415926")
  306.827 ns (0 allocations: 0 bytes)
3.1415926
```

```julia
julia> versioninfo()
Julia Version 1.7.3
Commit 742b9abb4d (2022-05-06 12:58 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: AMD Ryzen 7 5800H with Radeon Graphics
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-12.0.1 (ORCJIT, znver3)
Environment:
  JULIA_NUM_THREADS = auto
```

**Comment by quinnj** [comment]
Yeah, these kinds of issues were why I went ahead and built out a Julia-native float parsing code in Parsers.jl. Just weird inconsistencies between platforms w/ different settings and such. Not to mention it's slow in a lot of cases. Related #45301

**Comment by PallHaraldsson** [comment]
Doesn't this have a simple solution, use Parsers.jl, copy from there or depend on (as for DLMReader, see its closed issue, on bypassing Julia parsing)? Or better to use UCRT? Should this be marked as a "good first issue"?

We want Windows performance as good, also is the linked issue here a "correctness bug" (but not marked as such)? Making it more important to fix this issue/replace what's currently done?
